dashboard/app/helpers/levels_helper.rb

```diff
@@ -46,10 +46,6 @@ def readonly_view_options
   # Returns the channel associated with the given Level and User pair, or
   # creates a new channel for the pair if one doesn't exist.
   def get_channel_for(level, user = nil)
```

Comment:
The function level comment says we're creating a new channel if one doesn't exist, but in the case that user is not null, it looks like we are only searching for an existing token (in `ChannelToken.find_channel_token`), not creating a new one. It seems like we'd need to create a new one if one did not exist. Am I missing something?

Reply:
I think some of the confusion comes from differentiating between user and current_user. I'll try and make the comment better.

In the case where we're passed a user (not current_user), it means we have a teacher looking at a student's level. In this case we will never create a new channel.

In the case that we have a current_user, we will look for an existing channel_token and create one if it does not exist.

In the case that current_user is nil, the same thing will happen, except that `storage_id_for_user_id` will potentially end up generating a new storage_id for this anon user.